src/lib/keyseq.ts

```diff
@@ -515,3 +518,7 @@ browser.storage.onChanged.addListener(changes => {
         KEYMAP_CACHE = {}
     }
 })
+
+function updateBaseLayout() {
+    KEYCODETRANSLATEMAP = R.mergeRight(keyboardlayouts[config.get("keylayoutforcebase")], config.get("keylayoutforcemapping"))
+}
```

Comment:
Something like

```

function updateBaseLayout() {
    KEYCODETRANSLATEMAP = R.mergeRight(keyboardlayouts[config.get("keylayoutforcebase")], config.get("keylayoutforcemapping"))
}

setTimeout(() => config.addChangeListener("keylayoutforcebase", updateBaseLayout), 1000)
setTimeout(() => config.addChangeListener("keylayoutforcemapping", updateBaseLayout), 1000)
```

should work for invalidating but doesn't run at all and I can't work out why. (the `setTimeout`s are there because without them Tridactyl is just totally dead)